apis/python/src/tiledbsoma/soma_sparse_ndarray.cc

```diff
@@ -115,6 +115,28 @@ void load_soma_sparse_ndarray(py::module& m) {
         .def_property_readonly("shape", &SOMASparseNDArray::shape)
         .def_property_readonly("maxshape", &SOMASparseNDArray::maxshape)
         .def_property_readonly(
-            "has_upgraded_shape", &SOMAArray::has_current_domain);
+            "tiledbsoma_has_upgraded_shape", &SOMAArray::has_current_domain)
+
+        .def(
+            "resize",
+            [](SOMAArray& array, const std::vector<int64_t>& newshape) {
+                try {
+                    array.resize(newshape);
+                } catch (const std::exception& e) {
+                    throw TileDBSOMAError(e.what());
+                }
+            },
+            "newshape"_a)
+
+        .def(
+            "tiledbsoma_upgrade_shape",
+            [](SOMAArray& array, const std::vector<int64_t>& newshape) {
+                try {
+                    array.upgrade_shape(newshape);
+                } catch (const std::exception& e) {
+                    throw TileDBSOMAError(e.what());
+                }
+            },
+            "newshape"_a);
```

Comment:
I think we should do the rethrow in libtiledbsoma rather than pybind11. It'll simplify the code in here to be a one-liner again and the C++ API and R API would also consistently see `TileDBSOMAError` when erroring out from these methods.

Reply:
I wish that worked reliably! We found on #2980 and #2963 that, unfortunately, it does not. This has to be a catch-and-rethrow right here at the pybind layer, unfortunately. :(